File: #0590.n-ary-tree-postorder-traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<int> postorder(Node* root) {
        
        stack<Node*> stk;
        stk.push(root);
        vector<int> result;
        while (!stk.empty()) {
            Node* node = stk.top();
            stk.pop();
            
            if (node == NULL) {
                continue;
            }
            
            for (Node* child : node->children) {
                stk.push(child);
            }
            
            result.insert(result.begin(), node->val);
        }
        return result;
    }
};

File: #0590.n-ary-tree-postorder-traversal.js
/**
 * // Definition for a _Node.
 * function _Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {_Node|null} root
 * @return {number[]}
 */
var postorder = function(root) {
    const stack = root ? [root] : [];
    const results = [];
    while (stack.length) {
        root = stack.pop();
        if(root) {
            stack.push(...root.children);
            results.push(root.val);
        }
    }
    return results.reverse();
};
